src/hooks/useSwapEstimator.js

```diff
@@ -495,33 +473,63 @@ const useSwapEstimator = ({
 
   // Fetches current gas price
   const fetchGasPrice = async () => {
+    // https://docs.avax.network/learn/platform-overview/transaction-fees/#dynamic-fee-transactions
+    let jsonCallId = 1
+    let gasPriorityFee = BigNumber.from(0)
+    const nAVAX = '000000000'
+    let gasPrice = BigNumber.from('25' + nAVAX)
+
+    const fetchMethodResult = async (method) => {
+      const data = {
+        jsonrpc: '2.0',
+        method: method,
+        params: [],
+        id: jsonCallId.toString(),
+      }
+      jsonCallId++
+
+      const response = await fetch(process.env.ETHEREUM_RPC_PROVIDER, {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        referrerPolicy: 'no-referrer',
+        body: JSON.stringify(data),
+      })
+      if (response.ok) {
+        return get(await response.json(), 'result')
+      }
+    }
+
     try {
-      const gasPriceRequest = await fetchWithTimeout(
-        `https://ethgasstation.info/api/ethgasAPI.json?api-key=${process.env.DEFI_PULSE_API_KEY}`,
-        // allow for 5 seconds timeout before falling back to chainlink
-        {
-          timeout: 5000,
-        }
-      )
+      const gasPriceResponse = await fetchMethodResult('eth_baseFee')
+      if (gasPriceResponse.ok) {
+        gasPrice = BigNumber.from(
+          get(await gasPriceResponse.json(), 'result') + nAVAX
+        )
+      }
 
-      const gasPrice = BigNumber.from(
-        get(await gasPriceRequest.json(), 'average') + '00000000'
+      const gasPriorityFeeResponse = await fetchMethodResult(
+        'eth_maxPriorityFeePerGas'
       )
+      if (gasPriorityFeeResponse.ok) {
+        gasPriorityFee = BigNumber.from(
+          get(await gasPriorityFeeResponse.json(), 'result') + nAVAX
+        )
+      }
 
       if (!isGasPriceUserOverriden) {
         ContractStore.update((s) => {
           s.gasPrice = gasPrice
         })
       }
-      return gasPrice
     } catch (e) {
       console.error(
-        `Can not fetch gas prices, using chainlink as fallback method: ${e.message}`
```

Comment:
To be clear about reasoning: there is not chainlink gas price. We could use a 3rd party API for the gas price to keep parity with 2-method attempt, but I think given lower volatility of avax gas (it's almost always ~25 nAVAX), that we are OK.